extract_data.py

```diff
@@ -15,11 +15,11 @@
 
 
 def extract_and_collect_data_from_generated_files(directory: str, has_dedicated_gpu: bool, gpu_in_cpu: bool,
-                                                  verbose: bool = False, cleanup: bool = False):
+                                                  verbose: bool = False, gui: bool = True):
```

Comment:
I know it works anyway, but shouldn't the logic here be reversed, since the default is to not use the GUI?
```suggestion
                                                  verbose: bool = False, gui: bool = False):
```

Reply:
Well, the cleanup flag was exactly the same. False as default, True otherwise (which should be the "logic" default in this case). I went for this solution to avoid any change in main_with_gui.py. It felt handy in this case: I completely changed the input values in basically all functions of extract_data.py without touching main_with_gui (and keeping consistent results, obviously).
I thought this just as a different preference on what to do, defintely not a problem changing it. 
You tell me